### PR TITLE
chore: add issue and pull request templates

### PR DESCRIPTION
**Pull request by kushthedude**
Signed-off-by: Kush Trivedi <kushthedude@gmail.com>

/cc @oschaaf

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md
Comment (oschaaf) [inline .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md:12]: Maybe mention familiarizing oneself with https://github.com/envoyproxy/nighthawk/blob/master/CONTRIBUTING.md ?

**Review by oschaaf — CHANGES_REQUESTED** [review]
Thanks for adding this! 
I'm curious, is there a way to preview the effects of this before merging?
I have a suspicion I know the answer to that (no), but I thought lets ask anyway.

As for the substance of this PR, leaving one ask, other than that this LGTM.

I'll defer to @dubious90 for final review.

File: .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md
Comment (kushthedude) [reply .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md:12]: I guess you are right, also we can mention the DCO requirement I guess.

Comment (oschaaf) [reply .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md:12]: Good idea, yes. There's also pre-push/pre-commit hooks in https://github.com/envoyproxy/nighthawk/tree/master/support that may help with that.

Comment (dubious90) [reply .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md:12]: I'd actually argue further to Otto's point that I'd prefer we not duplicate where our contributing conventions are.  I'd prefer to only reference the other markdown file here rather than specify some conventions and reference it here.

I could probably be convinced in a different direction if you feel strongly.

**Review by dubious90 — COMMENTED** [review]
This looks awesome.  Thanks!  Just a couple suggestions

**Comment by oschaaf** [comment]
@kushthedude kindly pinging this PR, there is an open question in the review comments

File: .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md
Comment (kushthedude) [reply .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md:3]: We can add a comment saying if the PR is related to an issue then mention it below

File: .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md
Comment (kushthedude) [reply .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md:12]: Agree, we can reference the markdown file here than the other way around.

**Review by oschaaf — APPROVED** [review]
LGTM

**Review by dubious90 — APPROVED** [review]

**Comment by mum4k** [comment]
Ready for merging, blocked on the clang_tidy flake which we are planning to remove temporarily in the near future (#570).

**Comment by mum4k** [comment]
Please merge from master to remove the hard block on clang_tidy.

**Comment by kushthedude** [comment]
> Please merge from master to remove the hard block on clang_tidy.

Done @mum4k

**Pull request merged**
